### ISSUE TITLE
Question: Predicting one-step ahead using TFT

### ISSUE THREAD

**Issue by Azzeo** [jdb78/pytorch-forecasting#446]
I've been plucking my eyes out at this problem. And perhaps it's because I'm confused by how this package works.
Here's the deal. Let's assume we're attempting to predict demand as written in the [TFT tutorial ](https://pytorch-forecasting.readthedocs.io/en/latest/tutorials/stallion.html)(using the same steps). **Except** with a slightly different case: Predicting the time step ahead. While the validation set/test set  (last 6 months) is an out of sample that is used to benchmark other ML methods. In the aforementioned set, the rule for next time step ahead also applies. So to predict the last month of the test set, the volume data, etc is available of the second-last month etc... I am aware that one of the strengths of TFT is multi horizon forecasting, but my use case doesn't require this. If this scenario doesn't make sense to use TFT for, would another DL model be more appropriate (e.g. TabNet)? If it is appropriate. What vars do I need to adjust to make it work? Tinkering so far has resulted in errors such as "Unknown category '5' encountered"

**Comment by jdb78** [comment]
Not sure I understand the question but maybe you could just use `max_prediction_length=1` in the dataset?

**Comment by Azzeo** [comment]
> Not sure I understand the question but maybe you could just use `max_prediction_length=1` in the dataset?

That's what I tried! But then it would only predict 1 observation. How can I make it so that it predicts for the entire test set?

**Comment by arniwesth** [comment]
I guess the question is; how do you efficiently iterate over the (test) data set to predict 1 time step ahead for the entire set (which is something I have been wondering myself).

**Comment by jdb78** [comment]
By not using the `predict=True` keyword when constructing the dataset. It translates into taking the last available sequence per series.
